examples/common/lib/utils.sh

```diff
@@ -56,17 +56,40 @@ function wait_for_healthy_shard_primary() {
 	local wait_secs=180
 
 	for _ in $(seq 1 ${wait_secs}); do
-                if ! vtctldclient --server=localhost:15999 GetShard "${keyspace}/${shard}" | grep -qi "${unhealthy_indicator}"; then
+		if ! vtctldclient --server=localhost:15999 GetShard "${keyspace}/${shard}" | grep -qi "${unhealthy_indicator}"; then
 			break
 		fi
 		sleep 1
 	done;
 
-        if vtctldclient --server=localhost:15999 GetShard "${keyspace}/${shard}" | grep -qi "${unhealthy_indicator}"; then
+	if vtctldclient --server=localhost:15999 GetShard "${keyspace}/${shard}" | grep -qi "${unhealthy_indicator}"; then
 		fail "Timed out after ${wait_secs} seconds waiting for a primary tablet to be elected and become healthy in ${keyspace}/${shard}"
 	fi
 }
 
+
+# Wait for a primary tablet to be writeable, ie read_only=0 and super_read_only=0
+function wait_for_writeable_shard_primary() {
+	if [[ -z ${1} || -z ${2} ]]; then
+		fail "A keyspace and shard must be specified when waiting for the shard's primary to be healthy"
+	fi
+	local keyspace=${1}
+	local shard=${2}
+	local wait_secs=30
+
+	PRIMARY_TABLET="$(vtctldclient --server=localhost:15999 GetTablets --keyspace "$keyspace" --shard "$shard" | grep -w "primary" | awk '{print $1}')"
+	if [ -z "$PRIMARY_TABLET" ] ; then
+		fail "Cannot determine primary tablet for keyspace/shard $keyspace/$shard"
+	fi
+
+	for _ in $(seq 1 ${wait_secs}); do
+		if vtctldclient --server=localhost:15999 GetFullStatus "$PRIMARY_TABLET" | grep "super_read_only" | grep --quiet "false" ; then
+			break
+		fi
+		sleep 1
+	done;
```

Comment:
Shouldn't we fail if we did not find `super_read_only` after `wait_secs` has elapsed?

Reply:
done